### ISSUE TITLE
Bug? Sort-comp for propTypes and defaultProps broken as of node v8.x

### ISSUE THREAD

**Issue by RWOverdijk** [yannickcr/eslint-plugin-react#1906]
Our code contains invalid sorting order for propTypes and defaultProps. When running linting on CircleCI it correctly tells me:

```
...

/home/circleci/project/app/usage/components/UsageList.js
  18:3  error  defaultProps should be placed after propTypes  react/sort-comp

✖ 38 problems (38 errors, 0 warnings)
```

Running this on my machine (or any other dev's machine) doesn't complain about this at all and all rules pass.

I know that one of the rules is about the render method so I tested if this works for us: and it does. If I move render up in the class it tells me it's wrong:

```
103:3  error  render should be placed before constructor  react/sort-comp
```

~~I've compared the node version, npm version, node_modules (dev dep bin for eslint, airbnb version, eslint-plugin-react etc) versions. All identical.~~

It works as expected on Node version **v7.10.1** and doesn't work as expected on **v9.11.2**.

Also tried **v10.7.0**, and it doesn't work as expected.

**Comment by RWOverdijk** [comment]
One step further: If I make the static methods into instance methods, sort-comp picks them up.

After looking into the code it would appear that this happens because static methods go first. Still not sure how a different environment could affect this. Perhaps it has something to do with babel?

**Comment by RWOverdijk** [comment]
Another update, I was wrong. It is different with node versions. It works as expected on Node version **v7.10.1** and doesn't work as expected on **v9.11.2**.

Trying on v10.7.0 now.

**Comment by RWOverdijk** [comment]
For anyone coming across this:

For some reason, the newer versions of npm installed v7.7.0 of eslint-plugin-react. After upgrading that to v7.10.0 it worked.

[Looks like it was fixed here.](https://github.com/yannickcr/eslint-plugin-react/releases/tag/v7.9.0)

I found this by comparing sort-comp.js. If the top conditional does a static check, it's the wrong version.

Whew.

**Comment by ljharb** [comment]
Careful with lockfiles :-)
